chat/src/main/java/io/skygear/plugins/chat/UserConversation.java

```diff
@@ -20,102 +21,23 @@
     static final String LAST_READ_MESSAGE_KEY = "last_read_message";
 
     final Record record;
-    Message lastMessage;
-    Conversation conversation;
-
-    /**
-     * Instantiates a new user conversation relation from a Skygear Record.
-     *
-     * @param record the record
-     */
     UserConversation(@NonNull final Record record) {
         super();
-
         this.record = record;
-        this.lastMessage = null;
-        Object conversationObject = this.record.getTransient().get(CONVERSATION_KEY);
-        if (conversationObject == null) {
-            this.conversation = null;
-        } else {
-            this.conversation = new Conversation((Record) conversationObject);
-        }
-    }
-
-    /**
-     * Gets id.
-     *
-     * @return the id
-     */
-    @NonNull
-    public String getId() {
-        return this.record.getId();
-    }
-
-    /**
-     * Gets conversation.
-     *
-     * @return the conversation
-     */
-    @Nullable
-    public Conversation getConversation() {
-        return this.conversation;
-    }
-    public void setConversation(Conversation c) {
-        this.conversation = c;
     }
 
     /**
-     * Gets user.
+     * Instantiates a new conversation from a Skygear Record.
      *
-     * @return the user
-     */
-    @Nullable
-    public ChatUser getUser() {
-        Object chatUserObject = this.record.getTransient().get(USER_KEY);
-        if (chatUserObject == null) {
-            return null;
-        }
-
-        return new ChatUser((Record) chatUserObject);
-    }
-
-    /**
-     * Gets unread count.
-     *
-     * @return the unread count
+     * @param record the record
      */
-    public int getUnreadCount() {
-        return (int) this.record.get(UNREAD_COUNT_KEY);
-    }
 
-    /**
-     * Gets Last read message.
-     *
-     * @return the user
-     */
-    @Nullable
-    public Message getLastMessage() {
-        return this.lastMessage;
-    }
-
-    /**
-     * Gets last_read_message id.
-     *
-     * @return the message id without type
-     */
-    public String getLastReadMessageId() {
-        Reference ref = (Reference) this.record.get(LAST_READ_MESSAGE_KEY);
-        if (ref != null) {
-            return ref.getId();
-        }
-        return null;
-    }
-    /**
-     * Gets record.
-     *
-     * @return the Skygear record
-     */
-    public Record getRecord() {
-        return record;
+    @NonNull
+    static Conversation GetConversationByUserConversationRecord(@NonNull final Record record)
+    {
+        Object conversationObject = record.getTransient().get(CONVERSATION_KEY);
+        final int unreadCount = (int) record.get(UNREAD_COUNT_KEY);
+        final String lastReadMessageId = ((Reference) record.get(LAST_READ_MESSAGE_KEY)).getId();
```

Comment:
The LAST_READ_MESSAGE_KEY can be null. Currently it will throw NullPointerException